internal/target/repository.go

```diff
@@ -49,10 +58,11 @@ func NewRepository(r db.Reader, w db.Writer, kms *kms.Kms, opt ...Option) (*Repo
 		opts.WithLimit = db.DefaultLimit
 	}
 	return &Repository{
-		reader:       r,
-		writer:       w,
-		kms:          kms,
-		defaultLimit: opts.WithLimit,
+		reader:          r,
+		writer:          w,
+		kms:             kms,
+		defaultLimit:    opts.WithLimit,
+		listPermissions: opts.WithListPermissions,
```

Comment:
I think this should check that all of the permissions in `opts.WithListPemrissions` have `action.List` and `resource.Target`.